dataflow/operators/common.py

```diff
@@ -129,17 +129,31 @@ def fetch_from_api_endpoint(
     auth_token: Optional[str] = None,
     results_key: Optional[str] = "results",
     next_key: Optional[str] = "next",
+    basic_auth: Optional[bool] = False,
```

Comment:
I wonder if this might be better as this? Then the below block could be simplified to sub in this parameter in the f-string, rather than having two basically-identical blocks.

Speaking of nearly identical, this parameter also seems to control whether the `Content-Type` header is sent, which maybe is a bit obscure - I don't that content-type is directly related to basic auth, is it? So possibly that should also be a separate param, e.g. `extra_headers`? Not sure ... 

```suggestion
    auth_type: str = "Token",
```

Reply:
Oo yeah that sounds good! I knew there must be a more succinct way of doing it! 😋